src/link/link_info/bridge_port.rs

```diff
@@ -0,0 +1,581 @@
+const MDB_RTR_TYPE_TEMP_QUERY: u8 = 1;
+const MDB_RTR_TYPE_PERM: u8 = 2;
+const MDB_RTR_TYPE_TEMP: u8 = 3;
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+#[non_exhaustive]
+pub enum BridgePortState {
+    Disabled,
+    Listening,
+    Learning,
+    Forwarding,
+    Blocking,
+    Other(u8),
+}
+
+impl From<u8> for BridgePortState {
+    fn from(value: u8) -> Self {
+        use self::BridgePortState::*;
+        match value {
+            BR_STATE_DISABLED => Disabled,
+            BR_STATE_LISTENING => Listening,
+            BR_STATE_LEARNING => Learning,
+            BR_STATE_FORWARDING => Forwarding,
+            BR_STATE_BLOCKING => Blocking,
+            _ => Other(value),
+        }
+    }
+}
+
+impl From<BridgePortState> for u8 {
+    fn from(value: BridgePortState) -> Self {
+        use self::BridgePortState::*;
+        match value {
+            Disabled => BR_STATE_DISABLED,
+            Listening => BR_STATE_LISTENING,
+            Learning => BR_STATE_LEARNING,
+            Forwarding => BR_STATE_FORWARDING,
+            Blocking => BR_STATE_BLOCKING,
+            Other(v) => v,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+#[non_exhaustive]
+pub enum BridgePortMulticastRouter {
+    Disabled,
+    TempQuery,
+    Perm,
+    Temp,
+    Other(u8),
+}
+
+impl From<u8> for BridgePortMulticastRouter {
+    fn from(value: u8) -> Self {
+        use self::BridgePortMulticastRouter::*;
+        match value {
+            MDB_RTR_TYPE_DISABLED => Disabled,
+            MDB_RTR_TYPE_TEMP_QUERY => TempQuery,
+            MDB_RTR_TYPE_PERM => Perm,
+            MDB_RTR_TYPE_TEMP => Temp,
+            _ => Other(value),
+        }
+    }
+}
+
+impl From<BridgePortMulticastRouter> for u8 {
+    fn from(value: BridgePortMulticastRouter) -> Self {
+        use self::BridgePortMulticastRouter::*;
+        match value {
+            Disabled => MDB_RTR_TYPE_DISABLED,
+            TempQuery => MDB_RTR_TYPE_TEMP_QUERY,
+            Perm => MDB_RTR_TYPE_PERM,
+            Temp => MDB_RTR_TYPE_TEMP,
+            Other(v) => v,
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+#[non_exhaustive]
+pub enum InfoBridgePort {
+    State(BridgePortState),
+    Priority(u16),
+    Cost(u32),
+    Mode(bool),
+    Guard(bool),
+    Protect(bool),
+    FastLeave(bool),
+    Learning(bool),
+    UnicastFlood(bool),
+    ProxyARP(bool),
+    LearningSync(DefaultNla), // currently not used in kernel, unkown type
+    ProxyARPWifi(bool),
+    RootId((u16, [u8; 6])),
+    BridgeId((u16, [u8; 6])),
+    DesignatedPort(u16),
+    DesignatedCost(u16),
+    Id(u16),
+    No(u16),
+    TopologyChangeAck(bool),
+    ConfigPending(bool),
+    MessageAgeTimer(u64),
+    ForwardDelayTimer(u64),
+    HoldTimer(u64),
+    Flush,
+    MulticastRouter(BridgePortMulticastRouter),
+    Pad,
+    McastFlood(bool),
+    McastToUcast(bool),
+    VlanTunnel(bool),
+    BcastFlood(bool),
+    GroupFwdMask(u16),
```

Comment:
`GroupForwardMask`

Reply:
Looking to other files, like bridge.rs, human readable names are used, so I will change them.

But for this specific case, there is a GroupFwdMask in bridge.rs. Maybe I should use the same name here too?

Reply:
Yes. `GroupFwdMask` is OK in this case.